api/src/main/java/com/github/streamshub/console/api/model/HasLinks.java

```diff
@@ -0,0 +1,26 @@
+package com.github.streamshub.console.api.model;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+interface HasLinks<T> {
+
+    Map<String, String> links();
+
+    void links(Map<String, String> links);
+
+    default T addLink(String key, String value) {
+        links(addEntry(links(), key, value));
+        @SuppressWarnings("unchecked")
+        T t = (T) this;
+        return t;
+    }
+
+    private static <K, V> Map<K, V> addEntry(Map<K, V> map, K key, V value) {
+        if (map == null) {
+            map = new LinkedHashMap<>();
+        }
+        map.put(key, value);
```

Comment:
do we need to think about the case where an implementation is using a immutable map?

Reply:
Not in this case. We're in control of the implementations and there shouldn't be a need for them to be immutable. These are only used to build REST responses.